Vostok.Logging.File/EventsWriting/EventsWriterProvider.cs

```diff
@@ -53,7 +54,10 @@ public async Task<IEventsWriter> ObtainWriterAsync(CancellationToken cancellatio
                     var rollingStrategy = rollingStrategyProvider.ObtainStrategy();
 
                     var currentFile = rollingStrategy.GetCurrentFile(basePath);
-                    if (currentFile != cache.file || ShouldReopenWriter(cache.settings, settings) || cache.writer == null)
+                    if (currentFile != cache.file || 
+                        ShouldReopenWriter(cache.settings, settings) || 
+                        cache.writer == null || 
+                        (settings.FileShare.HasFlag(FileShare.Delete) && System.IO.File.Exists(currentFile.NormalizedPath) == false))
```

Comment:
Why it is not inside ShouldReopenWriter?

And this check should probably extracted to some well-named function (ShouldReopenDeletedFile for example).

Reply:
Resolved